### DISCUSSION TITLE
Concurrency vs Sandboxed vs Normal

### DISCUSSION THREAD

**Discussion by notflip** [taskforcesh/bullmq#1998] [Q&A]
I can't seem to wrap my head around these concepts, could someone explain to me what the difference is between these options?

- Using no (0) concurrency setting (does this mean every job is processed on the main thread?)
- Using only concurrency setting
- Sandboxed without concurrency setting
- Sandboxed with concurrency setting

### What if I have 2 queues and their workers, without sandboxing or concurrency, will these block each other from processing?

```
export const parseQueue = new Queue('parse', { connection });
export const parseWorker = new Worker('parse', async (job) => {});

export const planQueue = new Queue('plan', { connection });
export const planWorker = new Worker('plan', async (job) => {});
```

**Comment by manast** [comment]
The concurrency setting sets the number of jobs that can be processed in parallel by a given worker. It does not matter if the worker runs in a sandbox or in the main thread, it is the same concept.

**Reply by manast** [reply]
Btw, "without concurrency setting" is concurrency 1, the defaul.
